### PR TITLE
Fix INT8 export calibration ignoring `fraction` for classify models

### PR DESCRIPTION
**Pull request by synml**
## Problem

`fraction` is silently ignored when exporting a **classification** model with INT8 quantization — the full validation set is always used for calibration:

```python
from ultralytics import YOLO

YOLO("yolo26n-cls.pt").export(format="openvino", quantize=8, data="imagenet10", fraction=0.5)
# Before: calibrates on all 12 val images
# After:  calibrates on 6 val images
```

Detection/segmentation/pose/OBB models are unaffected — their branch passes `fraction` explicitly to `build_yolo_dataset()`.

## Root cause

Regression from #24972, which replaced `YOLODataset(..., fraction=self.args.fraction)` with `ClassificationDataset(..., augment=False)` in `Exporter.get_int8_calibration_dataloader()`. `ClassificationDataset` only applies `fraction` when `augment=True` (a deliberate gate so validation during training is never reduced), so the calibration path with `augment=False` skips it.

## Fix

Slice the calibration samples in the exporter's classify branch, mirroring what the detect branch already does by passing `fraction` to `build_yolo_dataset()`. The `augment` gate in `ClassificationDataset` is intentionally left untouched — removing it would make `fraction` shrink the val set during classification training.

## Impact scope

- Applies to **every INT8 export format** that builds its calibration set through the shared `Exporter.get_int8_calibration_dataloader()` (OpenVINO, ONNX, TensorRT, TF SavedModel, EdgeTPU, LiteRT, IMX, RKNN, Axelera, QNN, Hailo) — but **only** for `task=classify` with `fraction < 1.0`.
- No behavior change for the default `fraction=1.0`, for non-classify tasks, or for FP16/FP32 exports (no calibration set is built there).
- No change to training or validation dataset handling — `ClassificationDataset` itself is untouched.

## Verified

- imagenet10 (12 val images), `fraction=0.5` → log shows `found 6 images`, `Statistics collection 6/6`, export succeeds.
- Detect sanity check (coco8, `fraction=0.5` → 2 images) unchanged.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Fixes INT8 classification export calibration so the configured `fraction` value is respected.

### 📊 Key Changes
- Applies `self.args.fraction` to limit classification calibration samples when the fraction is below 1.0.
- Preserves the existing full-dataset behavior when `fraction` is 1.0 or higher.
- Keeps calibration preprocessing unchanged, including resizing, center cropping, and conversion to `uint8` tensors.

### 🎯 Purpose & Impact
- 🐛 Resolves a bug where classification INT8 exports ignored the requested calibration dataset fraction.
- ⚡ Allows users to reduce calibration data and potentially shorten export times and resource usage.
- ✅ Aligns classification calibration behavior with the configured export arguments without affecting other model types.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @synml, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! To ensure a seamless integration of your work, please review the following checklist:

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing). Don't hesitate to leave a comment if you have any questions. This review is automated, and an engineer will assist with any follow-up questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — APPROVED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Clean, focused fix. The classification calibration dataset is now reduced with the same rounded fraction semantics used by the detection dataset path, while preserving the inference-style calibration transforms and existing empty-dataset validation.

**Comment by glenn-jocher** [comment]
Please add a focused regression test in `tests/test_exports.py` that builds the classification calibration dataloader with `fraction=0.5` and asserts `len(dataloader.dataset) == 6` for `imagenet10`; this will lock in the fix and address the missing patch coverage.

**Comment by synml** [comment]
@glenn-jocher Done ✅ Added test_int8_calibration_classify_fraction to tests/test_exports.py — it builds the classification calibration dataloader directly with fraction=0.5 on imagenet10 and asserts len(dataloader.dataset) == 6 (12 val images × 0.5), following the same lightweight pattern as the existing test_int8_calibration_validates_split so no actual export runs in CI. Verified passing locally (~1.4s); reverting the fix makes it fail with 12 samples, so the regression is locked in and the missing patch coverage is addressed.
